src/frontend/src/eth/components/transactions/Transaction.svelte

```diff
@@ -2,43 +2,84 @@
 	import { nonNullish } from '@dfinity/utils';
 	import { BigNumber } from '@ethersproject/bignumber';
 	import type { ComponentType } from 'svelte';
-	import type { EthTransactionType } from '$eth/types/eth-transaction';
+	import type { Erc20Token } from '$eth/types/erc20';
+	import type { EthTransactionType, EthTransactionUi } from '$eth/types/eth-transaction';
+	import { isSupportedEthTokenId } from '$eth/utils/eth.utils';
 	import { isTransactionPending } from '$eth/utils/transactions.utils';
+	import IconBurn from '$lib/components/icons/IconBurn.svelte';
+	import IconConvert from '$lib/components/icons/IconConvert.svelte';
+	import IconMint from '$lib/components/icons/IconMint.svelte';
 	import IconReceive from '$lib/components/icons/IconReceive.svelte';
 	import IconSend from '$lib/components/icons/IconSend.svelte';
 	import TransactionPending from '$lib/components/transactions/TransactionPending.svelte';
 	import Amount from '$lib/components/ui/Amount.svelte';
 	import Card from '$lib/components/ui/Card.svelte';
 	import RoundedIcon from '$lib/components/ui/RoundedIcon.svelte';
-	import { ethAddress } from '$lib/derived/address.derived';
+	import { tokenId, tokenSymbol } from '$lib/derived/token.derived';
 	import { i18n } from '$lib/stores/i18n.store';
 	import { modalStore } from '$lib/stores/modal.store';
-	import type { Transaction } from '$lib/types/transaction';
+	import { token } from '$lib/stores/token.store';
 	import { formatSecondsToDate } from '$lib/utils/format.utils';
+	import { replacePlaceholders } from '$lib/utils/i18n.utils';
 
-	export let transaction: Transaction;
+	export let transaction: EthTransactionUi;
 
-	let from: string;
 	let value: BigNumber;
 	let timestamp: number | undefined;
 	let displayTimestamp: number | undefined;
+	let type: EthTransactionType;
 
 	let pending: boolean;
 	$: pending = isTransactionPending(transaction);
 
-	$: ({ from, value, timestamp, displayTimestamp } = transaction);
+	$: ({ value, timestamp, displayTimestamp, uiType: type } = transaction);
 
-	let type: EthTransactionType;
-	$: type = from?.toLowerCase() === $ethAddress?.toLowerCase() ? 'send' : 'receive';
+	let ckTokenSymbol: string;
+	$: ckTokenSymbol =
+		nonNullish($tokenId) && isSupportedEthTokenId($tokenId)
+			? `ck${$tokenSymbol}`
+			: ($token as Erc20Token).twinTokenSymbol ?? '';
```

Comment:
If `$tokenId` can potentially be nullish, couldn't `$token` be also potentially nullish?

Reply:
yes, that `nonNullish` is just because `isSupportedEthTokenId` does not accept `undefined`... what would you suggest?

Reply:
If the `$token` is potentially `undefined` then the following cast is maybe more accurate:

```
($token as Erc20Token | undefined)?.twinTokenSymbol ?? '';
```

Reply:
But if it cannot be undefined, it's fine.

Reply:
I will use your suggestion, but I add a TODO about changing it when `$token` will become something "better" eheh